src/extension.ts

```diff
@@ -0,0 +1,140 @@
+ * @param params params extracted from the document's URI.
+ * @param projects Sentry extension projects configurations.
+ * @return Sentry projectID this document reports to.
+ */
+export function matchSentryProject(params: Params, projects: Settings['sentry.projects']): SentryProject | undefined {
+    // Check if a Sentry project is associated with this document's repo and retrieve the project.
+    const project = projects!.find(p => !!new RegExp(p.patternProperties.repoMatch).exec(params.repo!))
+    if (!project) {
+        return
+    }
+
+    // Check if document matches the file matching pattern specified under the Sentry extension configuration.
+    const fileMatched: boolean = !!project.patternProperties.fileMatches.find(
+        pattern => !!new RegExp(pattern).exec(params.file!)
+    )
+    if (!fileMatched) {
+        return
+    }
+
+    return {
+        projectId: project.projectId,
+        lineMatches: project.patternProperties.lineMatches,
+    }
+}
+
+function decorateEditor(editor: sourcegraph.CodeEditor, sentryProjectId: string, lineMatches: RegExp[]): void {
+    const decorations: sourcegraph.TextDocumentDecoration[] = []
+    for (const [i, line] of editor.document.text!.split('\n').entries()) {
+        let m: RegExpExecArray | null
+        for (let pattern of lineMatches) {
+            pattern = new RegExp(pattern, 'gi')
+            do {
+                m = pattern.exec(line)
+                if (m) {
+                    decorations.push({
+                        range: new sourcegraph.Range(i, 0, i, 0),
+                        isWholeLine: true,
+                        after: {
+                            backgroundColor: '#e03e2f',
+                            color: 'rgba(255, 255, 255, 0.8)',
+                            contentText: ' View logs in Sentry » ',
+                            // Depending on the line matching pattern the query is indexed in position 1 or 2.
+                            linkURL: buildUrl(m.length > 2 ? m[2] : m[1], sentryProjectId).toString(),
+                        },
+                    })
+                }
+            } while (m)
+            pattern.lastIndex = 0 // reset
+        }
+    }
+    editor.setDecorations(DECORATION_TYPE, decorations)
+}
+
+/**
+ * Build URL to the Sentry issues stream page with the Sentry Org, Sentry project ID, and query.
+ * @param errorQuery extracted from the error handling code matching the config matching pattern.
+ * @param sentryProjectId from the associated Sentry project receiving logs from the document's repo.
+ * @return URL to the Sentry unresolved issues stream page for this kind of query.
+ */
+function buildUrl(errorQuery: string, sentryProjectId: string): URL {
+    const url = new URL(
+        'https://sentry.io/organizations/' +
+            encodeURIComponent(SETTINGSCONFIG['sentry.organization']!) +
+            '/issues/?project=' +
+            encodeURIComponent(sentryProjectId) +
```

Comment:
[`URLSearchParams`](https://developer.mozilla.org/en-US/docs/Web/API/URLSearchParams) would be helpful here.

Reply:
Will do this in the next PR.